### DISCUSSION TITLE
Restoring the SkeletonModificationStack3D feature

### DISCUSSION THREAD

**Discussion by Shnazzy** [godotengine/godot-proposals#7468] [3D]
The original SkeletonModificationStack3D was found to have been too unstable to include in the 4.0 release, so it was deprecated (https://github.com/godotengine/godot/pull/71137). 

Before it was deprecated, I had tried to submit a PR to fix part of it, but it wasn't able to be reviewed and accepted before the feature freeze.  (https://github.com/godotengine/godot/pull/61259).  I've since migrated the work I had in that PR over to a new branch that's kept up to date with master (https://github.com/Shnazzy/godot/tree/skeleton-mod-restore).

However, I've also gotten the sense that even without the stability issues, feelings are mixed about if the feature fits with the rest of Godot's design philosophy.  Before I make a PR of this new branch, I'd like to know what the current status of that conversation is.  If the stability issues were resolved, would the 3D skeleton modification stack be acceptable?

**Comment by yosoyfreeman** [comment]
Hi there! I personally believe that the Modification stack indeed seems like a piece of software made for another engine. A simple setup becomes a massive amount of nested panels because a single node is supposed to handle all the IK setup. Also, it only works with bones, so is not really a full IK solution. IK is specially useful to create chains inside the editor using objects. For example, assembling any kind of robotic arm is usually done in editor with modular pieces, and this can not be done, so my feeling is that the core functions are not there neither.  I think that one node should be equal to one IK modification chain, like physics constrains.

**Comment by Calinou** [comment]
> If the stability issues were resolved, would the 3D skeleton modification stack be acceptable?

Last time I remember, the consensus was that the feature itself needs to be redesigned.

**Comment by TokageItLab** [comment]
Closed by https://github.com/godotengine/godot/pull/87888
